ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -12488,8 +12489,9 @@ private void walkASTMarkTABREF(TableMask tableMask, ASTNode ast, Set<String> cte
     if (needRewritePrivObjs != null && !needRewritePrivObjs.isEmpty()) {
       for (HivePrivilegeObject privObj : needRewritePrivObjs) {
         MaskAndFilterInfo info = basicInfos.get(privObj);
-        // First we check whether entity actually needs masking or filtering
-        if (tableMask.needsMaskingOrFiltering(privObj)) {
+        // First we check whether entity actually needs masking or filtering. Query based Compaction related queries are
```

Comment:
what if someone triggers manual compaction? compaction type shouldn't matter

Reply:
Sorry, I don't get your point. The new session variable is added in QueryCompactor (in Worker), and this code also executes for every query. So regardless how the compaction was initiated the session variable will be there, and this code block will execute.

Could you please explain it a bit more?